packages/block-serialization-default-parser/parser.php

```diff
@@ -451,6 +451,7 @@ function next_token() {
 		 */
 		if ( $is_closer && ( $is_void || $has_attrs ) ) {
 			// we can ignore them since they don't hurt anything.
+			return array();
```

Comment:
It looks like this would change the way the parser works. Previously the token would be returned as a 'void-block'  (which I imagine is a block type without any inner content), but now the token is ignored. I think it could result in content in a user's post not being displayed.

It seems like the intention before was to avoid being particularly strict, and that's what the comment means when it says 'ignore them since they don't hurt anything'.

I think a reasonable option would be to remove the `if` statement and replace it with an explanatory comment like this:
```php
/*
 * A token could exist that matches this condition:
 *     $is_closer && ( $is_void || $has_attrs )
 *
 * This is ignored currently and treated as a void block, since it won't cause any issues.
 */
```

Reply:
Definitely also an option 👍🏻

Reply:
Returning an empty array results in a PHP warning:
```
Warning: Undefined array key 0 in parser.php on line 256
```
Also, the rest of the document is not being parsed.

In my opinion, this is not the desired intention.
I'd display an error message instead (if `WP_DEBUG` is enabled).
This would allow to:
a) Let the user know that the block markup is broken;
b) Don't stop parsing; the rest of the blocks will be parsed.

Reply:
P.S.
> I'd display an error message instead (if WP_DEBUG is enabled).

This could be explored in the follow-up PR.
For now, I'd just fix the PHPCS error (per @talldan's suggestion above) and call it a day.

Reply:
I'm not convinced about throwing an error. This particular error would be triggered when there's invalid markup in user content, and in particular the user would probably be mangling block markup by hand to achieve such an error.

Throwing an error effectively invalidates the entire user's document making it unsalvageable to the average user (there isn't a single try/catch in the file, and from what I can tell it handles all looping itself in the `parse` function). There's a good chance much of the document is still valid, so my opinion is that it makes sense to try to continue trying to parse the rest of the document. I would think of it as similar situation to how a web page will try to show a web page when there's incorrect markup.

What would an ideal fix look like? My thinking is that there would be some way to mark the particular token as invalid without invalidating the entire document, and also show these issues to users. This wouldn't be trivial, it'd involve developing a UI (both a block for visual editing mode, and a code highlighter for code editing mode) for displaying the errors.

What happens now in `trunk` isn't so bad, though it does make some assumptions in trying to fix the invalid content.